internal/cmd/controller/gitops/reconciler/gitjob_controller.go

```diff
@@ -103,9 +104,27 @@ func (r *GitJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		return ctrl.Result{}, fmt.Errorf("error retrieving git job: %v", err)
 	}
 
-	if errors.IsNotFound(err) && gitRepo.Status.Commit != "" {
-		if err := r.createJob(ctx, &gitRepo); err != nil {
-			return ctrl.Result{}, fmt.Errorf("error creating git job: %v", err)
+	if errors.IsNotFound(err) {
+		if gitRepo.Status.Commit != "" {
+			if err := r.createJob(ctx, &gitRepo); err != nil {
+				return ctrl.Result{}, fmt.Errorf("error creating git job: %v", err)
+			}
+		} else {
+			if gitRepo.Status.GitJobStatus == status.FailedStatus.String() {
+				return ctrl.Result{}, fmt.Errorf("error creating git job: %v", err)
+			}
+
```

Comment:
Why are we checking  for the `GitJobStatus` if the job was not found? 
Also... the log message seems wrong...

Reply:
Of course, I removed these lines